### DISCUSSION TITLE
why error using loadDatabases when internet is working

### DISCUSSION THREAD

**Discussion by PkiwiBird** [SysBioChalmers/GECKO#371] [Q&A] [answered]
When running `[ecModel, noUniprot] = makeEcModel(model, false);` there is an error:
```
Downloading Uniprot data for proteome UP000005640. This can take a few minutes.
Error using loadDatabases
Download failed, check your internet connection and try again, or manually
download:
https://rest.uniprot.org/uniprotkb/stream?query=reviewed:true+AND+proteome:UP000005640&fields=accession%2Cxref_geneid%2Cec%2Cmass%2Csequence&format=tsv&compressed=false&sort=protein_name%20asc

Error in makeEcModel (line 141)
uniprotDB = loadDatabases('both', modelAdapter);
```
The file can be saved manually, but since the internet is working fine, just would like to know if there could be another cause for this error?
And how can we get then the `ecModel` and `noUniprot` after saving that manually downloaded file as `uniprot.tsv` in folder data?
Because running the same line again after saving the file produced the same error.

**Comment by edkerk** [comment, marked as answer]
`makeEcModel` calls `loadDatabases`, which in turn checks if the file already exists locally, and should only download it if it does not exist.

`loadDatabases` will throw the error that you showed if the download fails for whatever reason. This could have to do with your internet connection, but it could also be a slow response by the UniProt server.

If you download the file manually, then make sure it is stored in the correct location:

```matlab
adapter = ModelAdapterManager.getDefault;
filepath = adapter.params.path;
fullfile(filepath,'data','uniprot.tsv')
```

If this file exists, then `loadDatabases` should not try to download it again next time.

I'll modify `loadDatabases` to provide this information in the error message.

**Reply by PkiwiBird** [reply]
thank you! I had an incorrect path set in the model adapter file.
